nengo_loihi/hardware/interface.py

```diff
@@ -105,8 +106,9 @@ def __enter__(self):
             raise SimulationError(
                 "Loihi interface has been closed and cannot be reopened."
             )
-        if not self.is_built:
-            self.build()
+
+        if not self.connected:
+            self.connect()
```

Comment:
So the reason I avoided this is because I didn't want to force the user to connect to the board when they do `with nengo_loihi.Simulator(...) as sim:`. One example of where you want to avoid this is when doing power measurement with a large model. Since you want to measure power as close to actual run time as possible, my process was something like this:
```
with nengo_loihi.Simulator(big_model) as sim:
    # big model has now built

    # measure idle power (involves connecting to the board, 
    # and I'm not sure if we can have two connections to the board at once)

    # run the big model
    sim.run(...)
```

Reply:
Not sure if this is relevant, but I was finding with the newer NxSDK's I could remove the `self.connect()` call and it would still automatically / lazily connect later on.

Reply:
So right now you can do this by making the sim first and opening the context manager later. As you pointed out in another comment, probably we should make it easier to do this without a context manager by adding an `open` method as well.

Reply:
Deferring entering the context manager is the way to do this now:

```
sim = nengo_loihi.Simulator(big_model)
# big model has now built

# measure idle power (involves connecting to the board, 
# and I'm not sure if we can have two connections to the board at once)

with nengo_loihi.Simulator(big_model) as sim:
    # run the big model
    sim.run(...)
```